Fix for ZipFileRenderer

WesGilster opened the pull request.

jmkao commented:
This doesn't fix the weird names on windows. That logic seems to be deeply buried inside of the githubReleases plugin, but it's mainly an aesthetic issue for dev builds, shouldn't affect stable releases since those will always be generated by Travis.

kloknibor commented:
I've got a question about this... It maybe be noob-ish. What if a printer manufacturer reales several printers, I can forsee they might like it better to just show theirs in this list. From what I understand the new script hardcodes the repo-owner on comipiling right? Is it still easy to just overwrite this file to set another place for getting the printer files? At the previous version I could just mount our.img and change the repo in this single file and done. Can this still be done easily?

WesGilster commented:
Your mixing up two different things. Community managed printers vs. build scripts. At the moment, community managed printer installs are only available in the dev stream and are hard coded to my repo. The fix you are looking for is to use the config.properties and stop hard coding. Print manufacturers should just use a custom property file and not touch the build.

WesGilster commented:
Speaking of builds, is everyone comfortable with the dev stream? Thinking about releasing to prod. Should I wait until start.sh is ready to pull from github releases?

WesGilster commented:
Your mixing up two different things. Community managed printers vs. build scripts. At the moment, community managed printer installs are only available in the dev stream and are hard coded to my repo. The fix you are looking for is to use the config.properties and stop hard coding. Print manufacturers should just use a custom property file and not touch the build.

WesGilster commented:
A little more "on topic". I like how this works now. Less that can go wrong and more control of the branch. Where would I put my grade.properties on Travis?

jmkao commented:
The Travis build doesn't rely on gradle.properties because that would require you to leave your token exposed in the repo somewhere.

Travis builds are controlled purely by `.travis.yml` using the Travis encryption mechanism for the credentials.

Travis only runs `gradlew clean dist` for its run, and then only in the area515 repo will it attempt to upload builds. In other repos (e.g. my repo, or your repo if you hook up travis), the upload will fail because the credentials will not work, but I think that's okay. With faster iterations in the dev forks, we probably can stick to using `gradlew githubRelease` for release uploads.

I think we're ready to release to prod. This will give us both `start.sh` and `newstart.sh` to try out, and then the only last step to fully remove all the binaries from the repo would be to replace `start.sh` with `newstart.sh`, which is a minor change. In order to make the migration seamless, we have to start propagating `newstart.sh` with the "old" binary-checkin way, and then switchover to `newstart.sh` so that older installs will still be abale to pull an update that takes them to the new way.

WesGilster committed:
Merge pull request #267 from WesGilster/master

Broke CWS printing without transforms

WesGilster committed:
Merge pull request #269 from WesGilster/master

Attempt to fix BLOCKING call

WesGilster committed:
Merge pull request #270 from WesGilster/master

Fixed slicer X/Y aspect ratio bugs related to X/Y pixel density

jmkao committed:
Fix link to RPi system image

jmkao committed:
Fix problem where imageFile is not initialized for ZipImagesFileProcessor.

The new logic for iterating up to an index in a customizer accidentally wiped out the normal initialization of the first image.

jmkao committed:
Fix some unnecessary import warnings

jmkao committed:
Fix areaMM computation for ZipImagesFileProcessor, but had to take out a cast to Boolean from RenderingCache, so check with Wes before pushing upstream.
